model/sharing/invitation.go

```diff
@@ -353,13 +355,58 @@ func (m *Member) SendShortcut(inst *instance.Instance, s *Sharing, link string)
 	return m.CreateSharingRequest(inst, s, creds, u)
 }
 
-// SendShortcutMail will send a notification mail after a shortcut for a
-// sharing has been created.
-func (s *Sharing) SendShortcutMail(inst *instance.Instance, fileDoc *vfs.FileDoc, previewURL string) error {
+func (s *Sharing) SendShortcutNotification(inst *instance.Instance, fileDoc *vfs.FileDoc, previewURL string) error {
 	sharerName := s.Members[0].PublicName
 	if sharerName == "" {
 		sharerName = inst.Translate("Sharing Empty name")
 	}
+	if err := s.SendShortcutPush(inst, fileDoc, previewURL, sharerName); err != nil {
+		inst.Logger().WithNamespace("sharing").
+			Warnf("Cannot send push notification: %s", err)
+	}
+	return s.SendShortcutMail(inst, fileDoc, previewURL, sharerName)
+}
+
+func (s *Sharing) SendShortcutPush(inst *instance.Instance, fileDoc *vfs.FileDoc, previewURL, sharerName string) error {
+	notifiables, err := oauth.GetNotifiables(inst)
+	if err != nil {
+		return err
+	}
+	hasFlashship := false
+	for _, notifiable := range notifiables {
+		if notifiable.Flagship {
```

Comment:
I had to manually add `flagship: true` to the oauth client couch document to make it work on my local setup.

When reading the code, I understand that `notifiable.Flagship` is true only for certified apps. It is a too strict check no ?